### PR TITLE
information: Fix issue of query `information_schema.columns` cost too much memory when there are lots of tables in TiDB

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

- No

### Release note 

- Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB.

### CODE REVIEW

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -1760,3 +1765,59 @@ func (e *memtableRetriever) setDataForStatementsSummary(ctx sessionctx.Context,
...
+	switch e.table.Name.O {
+	case infoschema.TableColumns:
+		err = e.setDataForColumns(sctx)
+	}
+	if err != nil {
+		return nil, err
+	}
+	e.retrieved = len(e.rows) == 0
+
+	//Adjust the amount of each return
```
Comment (tangenta) [inline executor/infoschema_reader.go:1806]:
```suggestion
```
This comment is incorrect here.

File: executor/infoschema_reader.go [hunk 2]
```diff
@@ -49,6 +49,20 @@ import (
...
 )
 
+func newMemTableRetriever(table *model.TableInfo, columns []*model.ColumnInfo) memTableRetriever {
+	switch table.Name.O {
+	case infoschema.TableColumns:
+		return &hugeMemTableRetriever{
+			table:   table,
+			columns: columns,
+		}
+	}
```
Comment (djshow832) [inline executor/infoschema_reader.go:59]: I think it's duplicated. Why not move this `switch-case` into `executor/builder.go` where you moved it out?

File: executor/infoschema_reader.go [hunk 3]
```diff
@@ -49,6 +49,20 @@ import (
...
+	return &memtableRetriever{
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply executor/infoschema_reader.go]: great, done.

**Pull request merged**